### PR TITLE
Toolbar new testrun

### PR DESCRIPTION
**Pull request by RMadjev**

### CODE REVIEW

File: tcms/testruns/templates/testruns/mutable.html [hunk 1]
```diff
@@ -74,8 +74,6 @@
         </div>
 
         {% if test_cases %}
-        {# Working around for the bug https://bugzilla.redhat.com/show_bug.cgi?id=715499 #}
-        <input type="hidden" name="POSTING_TO_CREATE" value="YES" />
```
Comment (RMadjev) [inline tcms/testruns/templates/testruns/mutable.html:78]: I want to pass all confirmed selected testcases as query params but obliviously there was a bug about that. Any suggestions?

File: tcms/testplans/static/testplans/js/get.js [hunk 2]
```diff
@@ -428,3 +440,17 @@ function getSelectedTestCases() {
     return tcIds;
 
 }
+
+function getConfirmedSelectedTestCases() {
```
Comment (atodorov) [inline tcms/testplans/static/testplans/js/get.js:444]: You can merge this together with getSelectedTestCases and return a complex data structure, see `selectedCheckboxes` in `refactor_testrun_ui` branch. However I'm questioning if this is at all necessary. 

The view creating new test run and its template is designed to warn the users if they've selected unconfirmed cases and there is already visual clues for that anyway. Independent of the fact if we have this function or not if the user tries to create a new TR by selecting unconfirmed cases it won't be possible.

File: tcms/testruns/templates/testruns/mutable.html (hunk 1 above)
Comment (atodorov) [reply tcms/testruns/templates/testruns/mutable.html:78]: This bug is "Can not create new test run with more than 700 cases" with the underlying root case:

"Request-URI Too Large,The requested URL's length exceeds the capacity limit for this server."

Which apparently is controlled by the [LimitRequestLine](http://httpd.apache.org/docs/2.4/mod/core.html#limitrequestline) httpd variable with a default of 8190 bytes.

Assuming 10 chars on average (`case=xxxx&`) that is 800 cases. You can try:
- using different query names, e.g. `p=` for plan, `c=` for case
- bumping this to 100KiB (should allow > 10K cases)
^^^ will need a test but that is way too extreme.  I would say 1000 to 2000 cases could be acceptable as the limit (e.g. 10KiB length of the URI) and then add additional cases afterwards.

Another option is keeping the selected cases in a cookie and reading that from the GET request which TBH seems better. 

FTR we've got `LimitRequestBody 10485760` which is 10 MiB for POST requests.

File: tcms/testruns/views.py [hunk 3]
```diff
@@ -36,25 +36,34 @@ class CreateTestRunView(View):
     """Display the create test run page."""
 
     template_name = 'testruns/mutable.html'
-    http_method_names = ['post']
+    http_method_names = ['post', 'get']
 
-    def post(self, request):
-        # If from_plan does not exist will redirect to plans for select a plan
-        plan_id = request.POST.get('from_plan')
+    def get(self, request):
```
Comment (atodorov) [inline tcms/testruns/views.py:41]: IMO the easiest way would be to split the view change into a separate pull request and make sure all tests referencing this view pass before continuing further.

**Review by atodorov — CHANGES_REQUESTED** [review]

File: tcms/testplans/static/testplans/js/get.js [hunk 4]
```diff
@@ -686,6 +686,18 @@ function toolbarEvents(testPlanId, permissions) {
 
          window.location.assign(`/cases/clone?case=${selectedCases.join('&case=')}`);
     });
+
+    $('#testplan-toolbar-newrun').click(function(ev) {
+        let selectedTestCases = getSelectedTestCases();
+
+        if (!selectedTestCases.length) {
+            alert($('#test_plan_pk').data('trans-no-confirmed-testcases-selected'));
```
Comment (atodorov) [inline tcms/testplans/static/testplans/js/get.js:694]: Wrong message for this check.

File: tcms/testplans/static/testplans/js/get.js [hunk 5]
```diff
@@ -686,6 +686,18 @@ function toolbarEvents(testPlanId, permissions) {
...
+            return;
+        }
```
Comment (atodorov) [inline tcms/testplans/static/testplans/js/get.js:696]: Also needs to check statuses of all the selected test cases!

File: tcms/testplans/templates/testplans/get.html [hunk 6]
```diff
@@ -25,6 +25,7 @@ <h1 class="col-md-12" style="margin-top: 0">
               data-trans-username-email-prompt="{% trans 'Enter username, email or user ID:' %}"
               data-trans-no-testcases-selected="{% trans 'No rows selected! Please select at least one!'%}"
               data-trans-are-you-sure="{% trans 'Are you sure?' %}"
+              data-trans-no-confirmed-testcases-selected="{% trans 'At least one confirmed case is required by a run.'%}"
```
Comment (atodorov) [inline tcms/testplans/templates/testplans/get.html:28]: Change to "Cannot create TestRun with unconfirmed test cases"

**Review by atodorov — CHANGES_REQUESTED** [review]

File: tcms/testplans/static/testplans/js/get.js (hunk 5 above)
Comment (RMadjev) [reply tcms/testplans/static/testplans/js/get.js:696]: Aren`t we discussed that already? We show a warning in create new run view, so the user is informed about that, and need to go back and reselect, also we should the statuses at create new run view.

Comment (atodorov) [reply tcms/testplans/static/testplans/js/get.js:696]: I don't remember but it is a relatively easy check and it can be added. Also the fact that there is a message in the HTML template suggests that we were thinking to add a simple check in the UI before potentially sending thousands of selected cases to the backend for examination.

File: tcms/testplans/static/testplans/js/get.js (hunk 4 above)
Comment (atodorov) [reply tcms/testplans/static/testplans/js/get.js:694]: :arrow_double_up:

File: tcms/testplans/templates/testplans/get.html [hunk 7]
```diff
@@ -25,6 +25,7 @@ <h1 class="col-md-12" style="margin-top: 0">
               data-trans-username-email-prompt="{% trans 'Enter username, email or user ID:' %}"
               data-trans-no-testcases-selected="{% trans 'No rows selected! Please select at least one!'%}"
               data-trans-are-you-sure="{% trans 'Are you sure?' %}"
+              data-trans-no-confirmed-testcases-selected="{% trans 'Cannot create TestRun with unconfirmed test cases'%}"
```
Comment (atodorov) [inline tcms/testplans/templates/testplans/get.html:28]: attribute name should match message or be similar at least.

File: tcms/testplans/static/testplans/js/get.js [hunk 8]
```diff
@@ -686,6 +686,26 @@ function toolbarEvents(testPlanId, permissions) {
 
          window.location.assign(`/cases/clone?case=${selectedCases.join('&case=')}`);
     });
+
+    $('#testplan-toolbar-newrun').click(function(ev) {
```
Comment (atodorov) [inline tcms/testplans/static/testplans/js/get.js:690]: `$(this).parents('.dropdown').toggleClass('open');` & `return false` down below. See lines 577-580

**Review by atodorov — CHANGES_REQUESTED** [review]
Minor change requested, otherwise looks good.

File: tcms/testplans/templates/testplans/get.html [hunk 9]
```diff
@@ -84,7 +85,7 @@ <h2 class="card-pf-title">
                 </h2>
 
                 <div class="card-pf-body" style="margin-top:0">
-                    {{ object.tree_view_html|safe }}
+{#                    {{ object.tree_view_html|safe }}#}
```
Comment (atodorov) [inline tcms/testplans/templates/testplans/get.html:88]: Revert

**Review by atodorov — CHANGES_REQUESTED** [review]
Looks good except the unrelated change.
